src/render/portable/mod.rs

```diff
@@ -866,16 +903,18 @@ impl JavascriptConfig {
             webview_host: webview_host.to_string(),
             is_single_page,
             page_size,
-            columns: columns.iter().map(|c| c.to_string()).collect(),
+            columns: columns.iter().map(|c| c.to_string()).chain(additional_columns.as_ref().unwrap_or(&HashMap::new()).keys().map(|c| c.to_string())).collect(),
```

Comment:
This and the added rows below look redundant to me. The chaining with additional columns given some filter reoccurs several times. Please move this into a closure that is just reinvoked here and below with different filters.

Reply:
Just moved the filtering of the display modes of the columns into a closure!